### PR TITLE
翻译 Tensor Attributes 和部分 oneflow.*

### PR DESCRIPTION
**Pull request by S882li**
[oneflow.device](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/framework/docstr/tensor_attributes.py#L19)
image
image

[oneflow.set_num_threads](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/framework/docstr/oneflow.py#L19)
image

[oneflow.vsplit](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/framework/docstr/math_ops.py#L1501)
image

[oneflow.t](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/framework/docstr/tensor_t.py#L19)
image

[oneflow.swapaxes](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/framework/docstr/swapaxes.py#L19)
image

[oneflow.select](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/framework/docstr/math_ops.py#L1290)
image

[oneflow.roll](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/nn/modules/roll.py#L20)
image

[oneflow.permute](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/framework/docstr/tensor_ops.py#L81)
image

[oneflow.ne](https://github.com/Oneflow-Inc/oneflow/blob/5c3a468836bc892749c26a1e1d6e5e9d3a27ad94/python/oneflow/framework/docstr/comparison.py#L108)
image

[oneflow.topk](https://github.com/Oneflow-Inc/oneflow/blob/5a66b1b8339a950be6e49649f4652233bfb0c61e/python/oneflow/framework/docstr/topk.py#L19)
image
image

[oneflow.tensor_split](https://github.com/Oneflow-Inc/oneflow/blob/5a66b1b8339a950be6e49649f4652233bfb0c61e/python/oneflow/framework/docstr/math_ops.py#L1418)
image

[oneflow.nms](https://github.com/Oneflow-Inc/oneflow/blob/5a66b1b8339a950be6e49649f4652233bfb0c61e/python/oneflow/framework/docstr/nms.py#L19)
image

### CODE REVIEW

File: docs/source/cn/tensor.py [hunk 1]
```diff
@@ -1249,4 +1249,58 @@
...
 
+reset_docstr(
+    oneflow.device,
+    r"""
+    该文档参考自：
+    https://pytorch.org/docs/stable/tensor_attributes.html#torch.torch.device
+
+    :class:`oneflow.device` 是一个对象，用于指代一个 :class:`oneflow.Tensor` 将被分配到的设备。
+
+    :class:`oneflow.device` 包括一个设备类型（ 'cpu' 或 'cuda' ）和一个可选的设备类型下的设备序号。如果设备序数不存在，则此对象将总是指代某设备类型下的当前设备。
```
Comment (rainyq69) [inline docs/source/cn/tensor.py:1260]:
```suggestion
    :class:`oneflow.device` 包括一个设备类型（ 'cpu' 或 'cuda' ）和一个可选的设备类型下的设备序号。如果设备序号不存在，则此对象将总是指代当前设备类型下的当前设备。
```

File: docs/source/cn/tensor.py [hunk 2]
```diff
@@ -1249,4 +1249,58 @@
...
+
+    :class:`oneflow.device` 中的设备可以通过 Tensor.device 属性获取。
```
Comment (rainyq69) [inline docs/source/cn/tensor.py:1262]:
```suggestion
    :class:`oneflow.device` 所指代的设备可以通过 Tensor.device 属性获取。
```

File: docs/source/cn/tensor.py [hunk 3]
```diff
@@ -1249,4 +1249,58 @@
...
+
+    :class:`oneflow.device` 可以通过一个 string 构建，也可以通过 string 和设备序号构建。
```
Comment (rainyq69) [inline docs/source/cn/tensor.py:1264]:
```suggestion
    :class:`oneflow.device` 可以通过一个 string 构建，也可以通过 string 和设备序号共同构建。
```

File: docs/source/cn/tensor.py [hunk 4]
```diff
@@ -1249,4 +1249,58 @@
...
+
+    通过 string 的构建方式：
```
Comment (rainyq69) [inline docs/source/cn/tensor.py:1266]:
```suggestion
    通过 string 构建的示例：
```

File: docs/source/cn/tensor.py [hunk 5]
```diff
@@ -1249,4 +1249,58 @@
...
+        >>> flow.device('cuda:0')
+        device(type='cuda', index=0)
+
+        >>> flow.device('cpu')
+        device(type='cpu', index=0)
+
+        >>> flow.device('cuda')  # 当前 cuda 设备
+        device(type='cuda', index=0)
+    
+    通过 string 和设备序号的构建方式：
```
Comment (rainyq69) [inline docs/source/cn/tensor.py:1280]:
```suggestion
    通过 string 和设备序号构建的示例：
```

File: docs/source/cn/constant.py [hunk 6]
```diff
@@ -402,3 +402,231 @@
...
+    .. WARNING::
+        为了保证正确数量的线程被使用， set_num_threads 必须在运行 eager ， eager globe 或 ddp 之前被调用。
+
+    """
+)
+
+reset_docstr(
+    oneflow.vsplit,
+    r"""
+    根据 indices_or_sections 的值，将输入的拥有两个或更高维度的张量垂直地分裂成多个张量。
```
Comment (rainyq69) [inline docs/source/cn/constant.py:420]:
```suggestion
    根据 indices_or_sections 的值，将输入的二维或更高维张量垂直地切分成多个张量。
```

File: docs/source/cn/constant.py [hunk 7]
```diff
@@ -402,3 +402,231 @@
...
+    每个分出的张量都是输入张量的 view 。此算子等价于调用 torch.tensor_split(input, indices_or_sections, dim=0) （分裂维度为 0 ），除了在 indices_or_sections 是
```
Comment (rainyq69) [inline docs/source/cn/constant.py:421]:
```suggestion
    每个切分出的张量都是输入张量的 view 。此算子等价于调用 torch.tensor_split(input, indices_or_sections, dim=0) （分裂维度为 0 ），除了在 indices_or_sections 是
```

Comment (rainyq69) [reply docs/source/cn/constant.py:421]: 官网的教程里写的是切分，统一一下吧。

File: docs/source/cn/constant.py [hunk 8]
```diff
@@ -402,3 +402,231 @@
...
+    一个整型的情况下，它必须整除分裂的维度，否则一个运行错误将被抛出。此文档参考自：https://pytorch.org/docs/stable/generated/torch.vsplit.html#torch.vsplit
```
Comment (rainyq69) [inline docs/source/cn/constant.py:422]:
```suggestion
    一个整型的情况下，此时它必须平均划分切分维度，否则将抛出一个运行时错误。此文档参考自：https://pytorch.org/docs/stable/generated/torch.vsplit.html#torch.vsplit
```

File: docs/source/cn/constant.py [hunk 9]
```diff
@@ -402,3 +402,231 @@
...
+    
+
+    参数：
+        - **input** (Tensor) - 输入张量
+        - **indices_or_sections** (int 或者一个列表) - 如果 indices_or_sections 是一个整型 n , input 将从维度 dim 被分裂成 n 个部分。
```
Comment (rainyq69) [inline docs/source/cn/constant.py:427]:
```suggestion
        - **indices_or_sections** (int 或者一个列表) - 如果 indices_or_sections 是一个整型 n , input 将沿着维度 dim 被切分为 n 个部分。
```

File: docs/source/cn/constant.py [hunk 10]
```diff
@@ -402,3 +402,231 @@
...
+            如果 input 的维度 dim 能被 n 整除，每个部分大小相同。如果 input 无法被 n 整除，最先的 int(input.size(dim) % n) 个部分将拥有 int(input.size(dim) / n) + 1 的
```
Comment (rainyq69) [inline docs/source/cn/constant.py:428]:
```suggestion
            如果在维度 dim 上 input 能被 n 整除，则每个部分大小相同都为 ``input.size(dim)/n``。如果 input 无法被 n 整除，前 int(input.size(dim) % n) 个部分（section）的大小都为 int(input.size(dim) / n) + 1，
```

File: docs/source/cn/constant.py [hunk 11]
```diff
@@ -402,3 +402,231 @@
...
+    oneflow.roll,
+    r"""将张量根据指定维度滚动。
+    
+    在末端位置的元素将被移动至首端位置。
+    
+    如果维度没有被指定，张量将在滚动前被扁平化，然后被恢复至原形状。
+
+    参数：
+        - **input** (oneflow.Tensor) - 输入张量
+        - **shifts** (int or int 元组 ) - 张量偏移的位置数量。如果 shifts 是一个元组，则 dims 必须也是一个相同大小的元组，且每个维度将会根据对应的值滚动。
```
Comment (rainyq69) [inline docs/source/cn/constant.py:547]:
```suggestion
        - **shifts** (int or int 元组 ) - 张量元素移动的位置数。如果 shifts 是一个元组，则 dims 必须也是一个相同大小的元组，且每个维度将会根据对应的值滚动。
```

File: docs/source/cn/constant.py [hunk 12]
```diff
@@ -402,3 +402,231 @@
...
+
+reset_docstr(
+    oneflow.permute,
+    r"""
+    permute(input, *dims) -> Tensor
+
+    返回原张量的维度换位后的 view 。
+
+    参数：
+        - **dims** (int 的元组 ) - 需要的维度顺序
```
Comment (rainyq69) [inline docs/source/cn/constant.py:587]:
```suggestion
        - **dims** (int 的元组 ) - 需要的维度排序
```

File: docs/source/cn/constant.py [hunk 13]
```diff
@@ -402,3 +402,231 @@
...
+
+reset_docstr(
+    oneflow.ne,
+    r"""ne(input, other) -> Tensor
+
+    计算 element-wise 的不等性。
+    第二个参数是一个可以与第一个参数广播的数字或张量。
+
+    参数：
+        - **input** (oneflow.Tensor): 用于比较的张量。
```
Comment (rainyq69) [inline docs/source/cn/constant.py:611]:
```suggestion
        - **input** (oneflow.Tensor): 需要比较的对象。
```

File: docs/source/cn/constant.py [hunk 14]
```diff
@@ -402,3 +402,231 @@
...
+        - **other** (oneflow.Tensor, float 或者 int): 用于比较的对象
```
Comment (rainyq69) [inline docs/source/cn/constant.py:612]:
```suggestion
        - **other** (oneflow.Tensor, float 或者 int): 与输入比较的张量。
```

File: docs/source/cn/constant.py [hunk 15]
```diff
@@ -402,3 +402,231 @@
...
+        oneflow.Size([2, 4, 5, 6])
+        >>> output[2].size()
+        oneflow.Size([1, 4, 5, 6])
+    """,
+)
+
+reset_docstr(
+    oneflow.t,
+    r"""
+    oneflow.t(input) → Tensor.
```
[the remark this answers is not in the document]
Comment (S882li) [reply docs/source/cn/constant.py]: https://github.com/Oneflow-Inc/oneflow-api-cn/pull/64

File: docs/source/cn/constant.py [hunk 16]
```diff
@@ -402,3 +402,231 @@
...
+
+        要求 `input` 小于二维，并交换第零维和第一维。
+
+        零维和一维张量将返回其本身。对于二维张量，此算子等价于 `transpose(input, 0, 1)` 。
+
+    参数：
+        - **input** (oneflow.Tensor) - 输入张量  
+ 
+    For example:
```
[the remark this answers is not in the document]
Comment (S882li) [reply docs/source/cn/constant.py]: 已解决

File: docs/source/cn/constant.py [hunk 17]
```diff
@@ -402,3 +402,230 @@
...
+    根据 indices_or_sections 的值，将输入的二维或更高维张量垂直地切分成多个张量。
+    每个切分出的张量都是输入张量的一个 view 。此算子等价于调用 torch.tensor_split(input, indices_or_sections, dim=0) （切分维度为 0 ），除了在 indices_or_sections 是
+    一个整型的情况下，此时它必须平均划分切分维度，否则将抛出一个运行时错误。此文档参考自：https://pytorch.org/docs/stable/generated/torch.vsplit.html#torch.vsplit
+    
+
+    参数：
+        - **input** (Tensor) - 输入张量
+        - **indices_or_sections** (int 或者一个列表) - 如果 indices_or_sections 是一个整型 n , input 将沿着维度 dim 被切分为 n 个部分。
+            如果在维度 dim 上 input 能被 n 整除，则每个部分的大小相同，都为 ``input.size(dim)/n``。如果 input 无法被 n 整除，则前 int(input.size(dim) % n) 个部分（section）的大小都为 int(input.size(dim) / n) + 1，
+            而余下的部分大小则是 int(input.size(dim) / n)。如果 indices_or_sections 是一个整型的列表或元组，则输入将从维度 dim 根据列表或元组内的每个索引被分裂。
```
Comment (rainyq69) [inline docs/source/cn/constant.py:429]:
```suggestion
            而余下的部分大小则是 int(input.size(dim) / n)。如果 indices_or_sections 是一个整型的列表或元组，则输入将在维度 dim 上根据列表或元组内的每个索引被切分。
```

File: docs/source/cn/constant.py [hunk 18]
```diff
@@ -402,3 +402,230 @@
...
+    每个切分出的张量都是输入张量的一个 view 。此算子等价于调用 torch.tensor_split(input, indices_or_sections, dim=0) （切分维度为 0 ），除了在 indices_or_sections 是
+    一个整型的情况下，此时它必须平均划分切分维度，否则将抛出一个运行时错误。此文档参考自：https://pytorch.org/docs/stable/generated/torch.vsplit.html#torch.vsplit
+    
+
+    参数：
+        - **input** (Tensor) - 输入张量
+        - **indices_or_sections** (int 或者一个列表) - 如果 indices_or_sections 是一个整型 n , input 将沿着维度 dim 被切分为 n 个部分。
+            如果在维度 dim 上 input 能被 n 整除，则每个部分的大小相同，都为 ``input.size(dim)/n``。如果 input 无法被 n 整除，则前 int(input.size(dim) % n) 个部分（section）的大小都为 int(input.size(dim) / n) + 1，
+            而余下的部分大小则是 int(input.size(dim) / n)。如果 indices_or_sections 是一个整型的列表或元组，则输入将在维度 dim 上根据列表或元组内的每个索引被切分。
+            比如， indices_or_sections=[2, 3] 和 dim=0 将使输入张量被分裂为 input[:2], input[2:3], 和 input[3:] 。如果 indices_or_sections 是一个张量，它必须是一个位于 cpu 的
```
Comment (rainyq69) [inline docs/source/cn/constant.py:430]:
```suggestion
            比如，若 indices_or_sections=[2, 3] 和 dim=0，输入张量将被切分为 input[:2], input[2:3], 和 input[3:]。如果 indices_or_sections 是一个张量，它必须是一个位于 cpu 的
```

**Review by rainyq69 — APPROVED** [review]

**Pull request merged**